src/ck-ldb/BaseLB.h

```diff
@@ -269,37 +269,29 @@ class LBVectorMigrateMsg : public CMessage_LBVectorMigrateMsg {
   LBVectorMigrateMsg(): level(0), n_moves(0) {}
 };
 
-// for a FooLB, the following macro defines these functions for each LB:
+// Deprecated, left around to support external custom load balancers
+// Internal load balancers should define lbinit directly and use LBRegisterBalancer<T>
+//
+// For a FooLB, the following macro defines these functions for each LB:
 // CreateFooLB():        create BOC and register with LBManager with a
 //                       sequence ticket,
 // AllocateFooLB():      allocate the class instead of a BOC
 // static void lbinit(): an init call for charm module registration
 #if CMK_LBDB_ON
 
-#define CreateLBFunc_Def(x, str)		\
-void Create##x(const CkLBOptions& opts) { 	\
-  CProxy_##x::ckNew(opts); 	\
-}	\
-\
-BaseLB *Allocate##x(void) { \
-  return new x((CkMigrateMessage*)NULL);	\
-}	\
-\
-static void lbinit(void) {	\
-  LBRegisterBalancer(#x,	\
-                     Create##x,	\
-                     Allocate##x,	\
-                     str);	\
-}
-
-#else		/* CMK_LBDB_ON */
-
-#define CreateLBFunc_Def(x, str)	\
-void Create##x(void) {} 	\
-BaseLB *Allocate##x(void) { return NULL; }	\
-static void lbinit(void) {}
-
-#endif		/* CMK_LBDB_ON */
+#  define CreateLBFunc_Def(x, str)                                                       \
```

Comment:
Should we consider adding explicit deprecation warnings here? Among potential solutions, we could generate a deprecated function as part of this macro, and call it within one of the other generated functions.

Reply:
Done.